js/pin.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+window.pin = (function () {
+  var pinElementTemplate = document.querySelector('#pin').content.querySelector('.map__pin');
+
+  return {
+    generatePins: function () {
+      var pinDocFrag = document.createDocumentFragment();
+      for (var i = 0; i < window.data.nearbyLisitings.length; i++) {
+        var newPinElement = pinElementTemplate.cloneNode(true);
+        var newImgElement = newPinElement.querySelector('img');
+        newPinElement.style.left = window.data.nearbyLisitings[i].location.x + 'px';
```

Comment:
create a variable `var nearbyLisiting = window.data.nearbyLisitings[i];`

and use it as `nearbyLisiting.location.x`, etc

Reply:
After moving the for loop to map.js I don't have a variable "i" so I pass to the function nearbyLisiting and current count of the for loop `window.pin.generatePin(pinData, i);`

Reply:
Why do you need to pass `i`? You need to pass the needed data to the generate, not the `i`.

Reply:
ok got it.